unified-runtime/source/adapters/level_zero/v2/command_list_manager.cpp

```diff
@@ -242,29 +242,36 @@ ur_result_t ur_command_list_manager::appendKernelLaunchUnlocked(
 ur_result_t ur_command_list_manager::appendKernelLaunch(
     ur_kernel_handle_t hKernel, uint32_t workDim,
     const size_t *pGlobalWorkOffset, const size_t *pGlobalWorkSize,
-    const size_t *pLocalWorkSize, uint32_t numPropsInLaunchPropList,
-    const ur_kernel_launch_property_t *launchPropList,
+    const size_t *pLocalWorkSize,
+    const ur_kernel_launch_ext_properties_t *launchPropList,
     uint32_t numEventsInWaitList, const ur_event_handle_t *phEventWaitList,
     ur_event_handle_t phEvent) {
   TRACK_SCOPE_LATENCY("ur_command_list_manager::appendKernelLaunch");
 
-  for (uint32_t propIndex = 0; propIndex < numPropsInLaunchPropList;
-       propIndex++) {
-    if (launchPropList[propIndex].id ==
-            UR_KERNEL_LAUNCH_PROPERTY_ID_COOPERATIVE &&
-        launchPropList[propIndex].value.cooperative) {
-      UR_CALL(appendKernelLaunchUnlocked(hKernel, workDim, pGlobalWorkOffset,
-                                         pGlobalWorkSize, pLocalWorkSize,
-                                         numEventsInWaitList, phEventWaitList,
-                                         phEvent, true /* cooperative */));
-      return UR_RESULT_SUCCESS;
-    }
-    if (launchPropList[propIndex].id != UR_KERNEL_LAUNCH_PROPERTY_ID_IGNORE &&
-        launchPropList[propIndex].id !=
-            UR_KERNEL_LAUNCH_PROPERTY_ID_COOPERATIVE) {
+  ur_kernel_launch_ext_properties_t *_launchPropList =
+      const_cast<ur_kernel_launch_ext_properties_t *>(launchPropList);
+  if (_launchPropList &&
+      _launchPropList->flags & UR_KERNEL_LAUNCH_FLAG_COOPERATIVE) {
+    UR_CALL(appendKernelLaunchUnlocked(
+        hKernel, workDim, pGlobalWorkOffset, pGlobalWorkSize, pLocalWorkSize,
+        numEventsInWaitList, phEventWaitList, phEvent, true /* cooperative */));
+    return UR_RESULT_SUCCESS;
+  }
+
+  if (_launchPropList &&
+      _launchPropList->flags & ~UR_KERNEL_LAUNCH_FLAG_COOPERATIVE) {
+    // We don't support any other flags.
+    return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+  }
+
+  while (_launchPropList != nullptr) {
```

Comment:
You can parse those properties using `find_stype_node` (like here https://github.com/intel/llvm/blob/sycl/unified-runtime/source/adapters/level_zero/usm.cpp#L1052).